frontend/src/components/dashboard/create/email/EmailTemplate.tsx

```diff
@@ -124,25 +168,63 @@ const EmailTemplate = ({
     return () => {
       setFinishLaterContent(null)
     }
-  }, [body, subject, from, setFinishLaterContent, campaignId, replyTo])
+  }, [
+    body,
+    subject,
+    fromName,
+    fromAddress,
+    setFinishLaterContent,
+    campaignId,
+    replyTo,
+  ])
 
   function replaceNewLines(body: string): string {
     return (body || '').replace(/<br\s*\/?>/g, '\n') || ''
   }
 
+  const handleSelectFromAddress = useCallback(
+    (selectedFrom: string) => {
+      const {
+        fromName: selectedFromName,
+        fromAddress: selectedFromAddress,
+      } = parseFromAddress(selectedFrom)
+
+      // Use custom from name if it has already been set. For e.g.,
+      // for "Custom <donotreply@mail.postman.gov.sg>"", we should
+      // use "Custom" instead of the default "Postman.gov.sg".
+      setFromName(
+        selectedFromAddress === initialFromAddress
+          ? initialFromName
+          : selectedFromName
+      )
+      setFromAddress(selectedFromAddress)
+    },
+    [initialFromName, initialFromAddress]
+  )
+
   return (
     <>
       <StepSection>
         <StepHeader title="Create email message" subtitle="Step 1" />
 
         <div>
           <h4>From</h4>
-          <p>Emails will be sent from this address</p>
+          <p>
+            Sender details. Custom sender name will be automatically appended
+            with &quot;via Government Mail&quot;.
```

Comment:
This should be 'via Postman' I think

Reply:
Maybe we can consider replacing this with a string interpolation using the constant `mailVia`

Reply:
Good catch!